src/blur/meta-blur-actor.c

```diff
@@ -0,0 +1,933 @@
+{
+    MetaBlurActorPrivate* priv = self->priv;
+    int scale = 2;
+
+    CoglContext *ctx = clutter_backend_get_cogl_context(clutter_get_default_backend());
+
+    int fb_width, fb_height;
+
+    ClutterActorBox box;
+    clutter_actor_get_allocation_box (CLUTTER_ACTOR (self), &box);
+
+    float width, height;
+    clutter_actor_box_get_size (&box, &width, &height);
+
+    fb_width = width;
+    fb_height = height;
+
+    fb_width = MAX (1, fb_width);
+    fb_height = MAX (1, fb_height);
+
+    fb_width >>= scale;
+    fb_height >>= scale;
+
+    if (priv->fbTex2 != NULL && (priv->fb_width == fb_width && priv->fb_height == fb_height)) {
+        return;
+    }
+
+    if (priv->fbTex2) {
+        g_clear_pointer (&priv->fbTex2, cogl_object_unref);
+        g_clear_pointer (&priv->fbTex, cogl_object_unref);
+        g_clear_pointer (&priv->fb, cogl_object_unref);
+        g_clear_pointer (&priv->fb2, cogl_object_unref);
+    }
+
+    priv->fb_width = fb_width;
+    priv->fb_height = fb_height;
+
+    priv->fbTex = cogl_texture_2d_new_with_size(ctx, priv->fb_width, priv->fb_height);
+    cogl_texture_set_components(priv->fbTex, COGL_TEXTURE_COMPONENTS_RGBA);
+    cogl_primitive_texture_set_auto_mipmap(priv->fbTex, FALSE);
+
+    CoglError *error = NULL;
+    if (cogl_texture_allocate(priv->fbTex, &error) == FALSE) {
+        meta_warning ("cogl_texture_allocat failed: %s\n", error->message);
+        goto _error;
+    }
+
+    priv->fb = cogl_offscreen_new_with_texture(priv->fbTex);
+    if (cogl_framebuffer_allocate(priv->fb, &error) == FALSE) {
+        meta_warning ("cogl_framebuffer_allocate failed: %s\n", error->message);
+        goto _error;
+    }
+
+    cogl_framebuffer_orthographic(priv->fb, 0, 0,
+            priv->fb_width, priv->fb_height, -1., 1.);
+
+    priv->fbTex2 = cogl_texture_2d_new_with_size(ctx, priv->fb_width, priv->fb_height);
+    cogl_texture_set_components(priv->fbTex2, COGL_TEXTURE_COMPONENTS_RGBA);
+    cogl_primitive_texture_set_auto_mipmap(priv->fbTex2, FALSE);
+
+    if (cogl_texture_allocate(priv->fbTex2, &error) == FALSE) {
+        meta_warning ("cogl_texture_allocat failed: %s\n", error->message);
+        goto _error;
+    }
+
+    priv->fb2 = cogl_offscreen_new_with_texture(priv->fbTex2);
+    if (cogl_framebuffer_allocate(priv->fb2, &error) == FALSE) {
+        meta_warning ("cogl_framebuffer_allocate failed: %s\n", error->message);
+        goto _error;
+    }
+
+    cogl_framebuffer_orthographic(priv->fb2, 0, 0,
+            priv->fb_width, priv->fb_height, -1., 1.);
+    return;
+
+_error:
+    g_clear_pointer (&priv->fbTex2, cogl_object_unref);
+    g_clear_pointer (&priv->fbTex, cogl_object_unref);
+    g_clear_pointer (&priv->fb, cogl_object_unref);
+    g_clear_pointer (&priv->fb2, cogl_object_unref);
+}
+
+static void preblur_texture(MetaBlurActor* self)
+{
+    MetaBlurActorPrivate *priv = self->priv;
+
+    float resolution[2] = {priv->fb_width, priv->fb_height};
```

Comment:
We could experiment with the resoltion a bit here. Making it gradually lower, we could get even bigger blur radius while improving peformance.